simbad/util/pdb_util.py

```diff
@@ -22,29 +22,34 @@ def __init__(self):
         self.hierarchy = None
         self.crystal_symmetry = None
 
-    def from_file(self, input_file):
+    @classmethod
+    def from_file(cls, input_file):
+        struct = cls()
         if input_file.endswith(".dat"):
-            pdb_str = read_dat(input_file)
-            self.pdb_input = iotbx.pdb.input(source_info=None, lines=pdb_str)
+            struct.pdb_input = iotbx.pdb.input(source_info=None, lines=read_dat(input_file))
         elif input_file.endswith(".pdb") or input_file.endswith(".ent"):
-            self.pdb_input = iotbx.pdb.pdb_input(file_name=input_file)
+            struct.pdb_input = iotbx.pdb.pdb_input(file_name=input_file)
         elif input_file.endswith(".ent.gz"):
             with gzip.open(input_file, "rb") as f_in:
```

Comment:
I don't think this line is needed and the line below should be indented once more